pkg/querier/worker/worker.go

```diff
@@ -232,6 +236,9 @@ func (w *querierWorker) AddressRemoved(address string) {
 // Must be called with lock.
 func (w *querierWorker) resetConcurrency() {
 	totalConcurrency := 0
+	defer func() {
+		w.metrics.concurrentWorkers.Set(float64(totalConcurrency))
```

Comment:
is there any additional overhead in using a defer function vs calling this on say line 274 before the if check? It seems that once the for loop it should be safe to set the value, but maybe this is for clarity (ie. express this is exported as soon as it's defined). I'm fine either way, just curious?

Reply:
No specific reason, I just like to set "clean up" logic beforehand (eat my veggies before dessert), but it's probably better to put it at the end and avoid the defer statement.

Reply:
🍨